storageconf: support YAML

RaduBerinde opened the pull request:
Add support for marshaling `storageconfig.Store` as yaml and parsing
it back. This is in support of switching from command-line flags to a
configuration file.

Size is now a tri-state that differentiates between unset or set to 0
bytes.

We also clean up some of the related code and move the validation of a
store config into `storageconfig`.

Epic: none
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/153827)
<!-- Reviewable:end -->

RaduBerinde committed:
storageconf: support YAML

Add support for marshaling `storageconfig.Store` as yaml and parsing
it back. This is in support of switching from command-line flags to a
configuration file.

Size is now a tri-state that differentiates between unset or set to 0
bytes.

We also clean up some of the related code and move the validation of a
store config into `storageconfig`.

Epic: none
Release note: None
